doi/src/main/java/ca/nrc/cadc/doi/PostAction.java

```diff
@@ -463,11 +474,78 @@ private void performDoiAction() throws Exception {
             log.debug("redirectUrl: " + redirectUrl);
             syncOutput.setHeader("Location", redirectUrl);
             syncOutput.setCode(303);
+        } else if (doiAction.equals(SEARCH_ACTION)) {
+            if (doiStatusSearchFilter == null || callersNumericId == null) {
+                getStatusList(getOwnedDOIList());
+            } else {
+                getStatusList(getFilteredDOIList(doiStatusSearchFilter));
+            }
         } else {
             throw new UnsupportedOperationException("DOI action not implemented: " + doiAction);
         }
     }
-    
+
+    private List<Node> getFilteredDOIList(DoiStatusSearchFilter doiStatusSearchFilter) throws Exception {
+        List<Node> filteredNodes = new ArrayList<>();
+        ContainerNode doiRootNode = vospaceDoiClient.getContainerNode("");
+
+        if (doiRootNode != null) {
+            for (Node childNode : doiRootNode.getNodes()) {
+                NodeProperty requester = childNode.getProperty(DOI_VOS_REQUESTER_PROP);
+                if (requester == null || requester.getValue() == null) {
+                    continue; // Skip nodes without a valid requester
+                }
+
+                // Check status filter
+                if (!doiStatusSearchFilter.getStatusList().isEmpty()) {
+                    NodeProperty statusProp = childNode.getProperty(DOI_VOS_STATUS_PROP);
+                    if (statusProp == null || !doiStatusSearchFilter.getStatusList()
+                            .contains(Status.toValue(statusProp.getValue()))) {
+                        continue; // Skip nodes that don't match the status filter
+                    }
+                }
+
+                // Check role filter
+                if (doiStatusSearchFilter.getRole() != null) {
+                    if (callersNumericId == null) {
```

Comment:
At this point, shouldn't be possible for callerNumericId to be null. The caller must be authenticated to do a post.